PF/Searching/Searching.java

```diff
@@ -0,0 +1,34 @@
+
+public class Searching {
+    public int linearSearch(int array[], int startIndex, int arrayLength, int valueToBeFind){
+        if(startIndex >= arrayLength){
+            return -1;
+        }
+        else{
+            if(array[startIndex] == valueToBeFind){
+                return startIndex;
+            }
+            else{
+                return linearSearch(array, startIndex+1, arrayLength, valueToBeFind);
+            }
+        }
+    }
+    
+    public int binarySearch(int array[], int startIndex, int lastIndex, int valueToBeFind){
+        if(startIndex > lastIndex){
+            return -1;
+        }
+        else{
+            int middleIndex = (startIndex + lastIndex)/2;
```

Comment:
variable declaration should not be in the recursion function, it should be class level